### PR TITLE
feat: refactor binary interactions in test framework

### PR DESCRIPTION
**Pull request by bharathkkb**
Consolidates test framework interaction with binaries into a binary pkg

### CODE REVIEW

File: infra/blueprint-test/pkg/gcloud/gcloud.go [hunk 1]
```diff
@@ -20,24 +20,23 @@ package gcloud
...
 	"github.com/gruntwork-io/terratest/modules/shell"
 	"github.com/mitchellh/go-testing-interface"
 	"github.com/tidwall/gjson"
 )
 
 type CmdCfg struct {
-	gcloudBinary string         // path to gcloud binary
-	commonArgs   []string       // common arguments to pass to gcloud calls
-	logger       *logger.Logger // custom logger
+	*binary.BinaryCfg                       // binary config
```
Comment (morgante) [inline infra/blueprint-test/pkg/gcloud/gcloud.go:30]: What's the reason to make this a pointer vs embedding the struct entirely?

File: infra/blueprint-test/pkg/binary/bin.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		opt(b)
+	}
+	err := utils.BinaryInPath(b.binary)
+	if err != nil {
+		t.Fatalf("unable to find binary %s in path: %v", b.binary, err)
+	}
+	return b
+}
+
+func (b *BinaryCfg) GetBinary() string {
```
Comment (morgante) [inline infra/blueprint-test/pkg/binary/bin.go:48]: We probably shouldn't use getters/setters. If these properties should be public, simply export them.

File: infra/blueprint-test/pkg/gcloud/gcloud.go [hunk 3]
```diff
@@ -20,24 +20,23 @@ package gcloud
...
+	bOpts             []binary.BinaryOption // binary options
+	commonArgs        []string              // common arguments to pass to gcloud calls
 }
 
 type cmdOption func(*CmdCfg)
 
-func WithBinary(gcloudBinary string) cmdOption {
+func WithBinaryOptions(bOpts ...binary.BinaryOption) cmdOption {
```
Comment (morgante) [inline infra/blueprint-test/pkg/gcloud/gcloud.go:37]: We shouldn't need this layer of indirection. As a consumer of the `gcloud` package, I shouldn't need to know if options are common (across all binaries) or specific to gcloud.

**Review by morgante — CHANGES_REQUESTED** [review]

File: infra/blueprint-test/pkg/gcloud/gcloud.go (hunk 3 above)
Comment (bharathkkb) [reply infra/blueprint-test/pkg/gcloud/gcloud.go:37]: @morgante in that case should we continue to keep the `WithBinary` and other options within the `gcloud` pkg?

Comment (morgante) [reply infra/blueprint-test/pkg/gcloud/gcloud.go:37]: @bharathkkb No, I think it should still be possible to share the options by using a interfaces. If this doesn't work I can give it a try.

Comment (bharathkkb) [reply infra/blueprint-test/pkg/gcloud/gcloud.go:37]: Thanks, I'll try that approach

**Comment by morgante** [comment]
@bharathkkb What's the status of this PR?

**Comment by bharathkkb** [comment]
@morgante I de prioritized this for some feature work. I will pick this up soon.

**Pull request closed without merging**